packages/x-components/src/x-modules/url/store/getters/url-params.getter.ts

```diff
@@ -17,10 +18,41 @@ export const urlParams: UrlXStoreModule['getters']['urlParams'] = ({
   return reduce(
     { ...rawUrlParams, ...extraParams },
     (urlParams, key, value) => {
-      const newKey = config.urlParamNames[key] ?? key;
-      urlParams[newKey] = value as UrlParamValue;
+      if (isValidParam(value as string | Array<string>) && !isFirstPage(key, value as number)) {
```

Comment:
- This if can grow really big, and I don't think it will serve us for every case.  For example, right now you can't save `0` or `false` values to the URL, which might be a valid use case.
- Another code smell is that you are doing 2 casts that should be unnecessary.
- And finally, conceptually is not a `reduce`, this is a `filter`. 

With all of that, there goes my proposal:

```
export const urlParams: UrlXStoreModule['getters']['urlParams'] = ({
  config,
  extraParams,
  ...rawUrlParams
}) =>  objectFilter({ ...rawUrlParams, ...extraParams }, isValidParam);

const invalidUrlValues: Record<string, UrlParamValue> = {
  page: 1
};

function isValidParam(key: string, value: UrlParamValue): boolean {
  return Array.isArray(value)
    ? value.length > 0
    : value != null && value !== '' && value !== invalidUrlValues[key];
}
```